crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -116,236 +114,191 @@ impl AsRef<[Variable]> for LightClientStateVar {
-            schnorr_ver_key_comm,
-            lightclient_state_var.stake_table_comm().schnorr_keys_comm,
-        )?;
-
-        // checking the commitment for the list of stake amounts
-        let stake_amount_preimage_vars = stake_table_var
-            .iter()
-            .map(|var| var.stake_amount)
-            .collect::<Vec<_>>();
-        let stake_amount_comm = RescueNativeGadget::<F>::rescue_sponge_with_padding(
-            &mut circuit,
-            &stake_amount_preimage_vars,
-            1,
-        )?[0];
-        circuit.enforce_equal(
-            stake_amount_comm,
-            lightclient_state_var.stake_table_comm().stake_amount_comm,
-        )?;
-
-        // checking all signatures
-        let verification_result_vars = stake_table_var
-            .iter()
-            .zip(sig_vars)
-            .map(|(entry, sig)| {
-                SignatureGadget::<_, P>::check_signature_validity(
-                    &mut circuit,
-                    &entry.schnorr_ver_key,
-                    lightclient_state_var.as_ref(),
-                    &sig,
-                )
-            })
-            .collect::<Result<Vec<_>, CircuitError>>()?;
-        let bit_x_result_vars = signer_bit_vec_var
-            .iter()
-            .zip(verification_result_vars)
-            .map(|(&bit, result)| {
-                let neg_bit = circuit.logic_neg(bit)?;
-                circuit.logic_or(neg_bit, result)
-            })
-            .collect::<Result<Vec<_>, CircuitError>>()?;
-        let sig_ver_result = circuit.logic_and_all(&bit_x_result_vars)?;
-        circuit.enforce_true(sig_ver_result.0)?;
-
-        circuit.finalize_for_arithmetization()?;
-        Ok((circuit, public_inputs))
+        })
+        .collect::<Result<Vec<_>, CircuitError>>()?;
+    let dummy_ver_key_var = VerKeyVar(circuit.neutral_point_variable());
+    stake_table_var.resize(
+        STAKE_TABLE_CAPACITY,
+        StakeTableEntryVar {
+            schnorr_ver_key: dummy_ver_key_var,
+            stake_amount: 0,
+        },
+    );
+
+    // creating variables for signatures
+    let mut sig_vars = sigs
+        .iter()
+        .map(|sig| circuit.create_signature_variable(sig))
+        .collect::<Result<Vec<_>, CircuitError>>()?;
+    sig_vars.resize(
+        STAKE_TABLE_CAPACITY,
+        SignatureVar {
+            s: circuit.zero(),
+            R: circuit.neutral_point_variable(),
+        },
+    );
+
+    // creating Boolean variables for the bit vector
+    let mut signer_bit_vec_var = signer_bit_vec
+        .iter()
+        .map(|&b| circuit.create_boolean_variable(b))
+        .collect::<Result<Vec<_>, CircuitError>>()?;
+    signer_bit_vec_var.resize(STAKE_TABLE_CAPACITY, BoolVar(circuit.zero()));
+
+    let threshold = u256_to_field::<F>(threshold);
+    let threshold_var = circuit.create_public_variable(threshold)?;
+
+    let lightclient_state_var = LightClientStateVar::new(&mut circuit, lightclient_state)?;
```

Comment:
Maybe rename it to `lightclient_state_pub_input_var` to remind the user that they are public input variables.

Reply:
Done in 3c40ff1